starter-code/routes/drones.js

```diff
@@ -1,21 +1,47 @@
-const express = require('express');
+const express = require("express");
 
 // require the Drone model here
+const Drone = require("../models/drone");
 
 const router = express.Router();
 
-
-router.get('/drones', (req, res, next) => {
+router.get("/", (req, res, next) => {
   // Iteration #2
+  Drone.find({}, (err, drones) => {
+    if (err) {
+      return next(err);
+    }
+
+    res.render("drones/index", {
+      drones: drones
+    });
+  });
 });
 
-
-router.get('/drones/new', (req, res, next) => {
+router.get("/new", (req, res, next) => {
   // Iteration #3
+  res.render("drones/new");
 });
 
-router.post('/drones', (req, res, next) => {
+router.post("/", (req, res, next) => {
   // Iteration #3
+  router.post("/", (req, res, next) => {
+    const droneInfo = {
```

Comment:
By the way, you can use clever ES6 syntax to deconstruct the body object and and the shorthand object prop syntax to create the `droneInfo` object, like this:


`const {name, propellers, maxSpeed} = req.body // this will create three variables `
`const droneInfo = {name, propellers, maxSpeed} // this is equivalent to {name: name ... etc}`